### PR TITLE
[MTDSA-6563] ignore employment expenses controller, route & ITs

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+          override def setupStubs(): StubMapping = {
+            AuditStub.audit()
+            AuthStub.authorised()
+            MtdIdLookupStub.ninoFound(nino)
+          }
+
+          val response: WSResponse = await(request().put(requestBody))
+          response.status shouldBe BAD_REQUEST
+          response.json shouldBe Json.toJson(TaxYearFormatError)
```
Comment (Mbrookesy) [inline it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala:132]: It seems to me at least the ISpec is testing for the FORMAT_NINO error and FORMAT_TAX_YEAR error twice, one individually and one in the Seq

File: it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+          (BAD_REQUEST, "INVALID_TAX_YEAR", NOT_FOUND, NotFoundError),
+          (BAD_REQUEST, "INVALID_CORRELATIONID", INTERNAL_SERVER_ERROR, DownstreamError),
+          (BAD_REQUEST, "INVALID_PAYLOAD", INTERNAL_SERVER_ERROR, DownstreamError),
+          (UNPROCESSABLE_ENTITY, "INVALID_REQUEST_BEFORE_TAX_YEAR_END", FORBIDDEN, RuleTaxYearNotEndedError),
```
Comment (Mbrookesy) [inline it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala:197]: Just a comment, seems a little odd that on the ignore endpoint RULE_TAX_YEAR_NOT_ENDED is status 403 while on amend it is 400

File: test/v1/controllers/IgnoreEmploymentExpensesControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    with MockIgnoreEmploymentExpensesService
+    with MockIgnoreEmploymentExpensesRequestParser
+    with MockHateoasFactory
+    with MockAuditService {
+
+
+  private val nino = "AA123456A"
+  private val taxYear = "2019-20"
+  private val correlationId = "X-123"
+  private val testHateoasLink = Link(href = s"individuals/expenses/other/$nino/$taxYear", method = GET, rel = "self")
```
Comment (Mbrookesy) [inline test/v1/controllers/IgnoreEmploymentExpensesControllerSpec.scala:49]: test uses the other expenses link

File: it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala (hunk 2 above)
Comment (james-work-account) [reply it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala:197]: Good point, I'll chase this up.

Comment (james-work-account) [reply it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala]: You're right, it should've been 400.

**Pull request merged**